app/models/recipe.rb

```diff
@@ -1,6 +1,7 @@
 class Recipe < ApplicationRecord
   belongs_to :user
-  has_many :foods
+  has_many :recipe_foods, dependent: :destroy
+  has_many :foods, through: :recipe_foods
 
   validates :name, :preparation_time, :cooking_time, :description, :public, presence: true
```

Comment:
-Optional: Hi @Shisui6 everything seems to work like a charm but it seems when I tried to create a recipe I have to make it public first then go and change it from the edit page, kindly let's make it flexible so that any user can create public and private recipes.
![Screenshot from 2023-03-15 09-06-26](https://user-images.githubusercontent.com/72081991/225226861-8ba72194-6fbd-4aca-9791-a29587849ad4.png)
#### Hint: We can remove `public` from recipe validation and add also give `read` privilege for the user in the `ability.rb` and lastly add default value of false in our migration.